### PR TITLE
Carer connection details

### PR DESCRIPTION
**Pull request by katbow**
For epic #90. Architecture change from phoenix app to umbrella app containing healthlocker & oxleas. Healthlocker portion contains phoenix logic, and oxleas contains some lib functionality.

#99 
Creates modules to:
* update all clinicians for a service user, create a room, and add service user & clinician to that room
* create carer/service user relationship, create a room, and add carer & clinician for the service user to that room

#100
Adds all routes into router for healthlocker/oxleas functionality. All routes that are oxleas specific are in scopes that say `Healthlocker.OxleasAdhd` as a way of clearly separating the generic vs specific routes. Any routes that are generic are under `Healthlocker`.

#103 
Some controllers that already existed in `Healthlocker` have different functionality in `OxleasAdhd`. These controllers were edited to fit the Oxleas functionality, and placed in a folder in `web/controllers/oxleas_adhd`, which also contains new controllers that did not exist in Healthlocker.

The views, templates, and lib functionality which is specific to oxleas follows the same folder structure as above for controllers. This is to help make the app more generic moving forward.

### CODE REVIEW

**Comment by katbow** [comment]
This is going to be assigned to @simonlab once he's accepted the invitation to collaborate. Didn't forget about assigning!

File: .travis.yml [hunk 1]
```diff
@@ -3,17 +3,31 @@ sudo: required
 language: elixir
 elixir:
 - 1.4.0
-cache:
-  directories:
-    - _build
-    - deps
-    - "travis_phantomjs"
+# cache:
```
Comment (SimonLab) [inline .travis.yml]: should we delete the commented lines if they are not used anymore?

File: apps/healthlocker/config/prod.exs [hunk 2]
```diff
@@ -11,20 +11,41 @@ use Mix.Config
...
-  cache_static_manifest: "priv/static/manifest.json"
+config :healthlocker, Healthlocker.Endpoint,
+  server: true, # see: https://stackoverflow.com/a/44855391/1148249
+  root: ".",
+  version: Mix.Project.config[:version],
+  http: [port: 4000],
+  debug_errors: true,
+  url: [host: "localhost"],
+  # http: [port: System.get_env("PORT")],
+  # url: [scheme: "http", host: System.get_env("HEROKU_URL")], # port: 443],
```
Comment (SimonLab) [inline apps/healthlocker/config/prod.exs]: Any reason why we don't use anymore the environment variables on here?

File: apps/healthlocker/config/prod.exs [hunk 3]
```diff
@@ -11,20 +11,41 @@ use Mix.Config
...
+  # force_ssl: [rewrite_on: [:x_forwarded_proto]],
+  # cache_static_manifest: "priv/static/manifest.json",
+  secret_key_base: System.get_env("SECRET_KEY_BASE")
+
+# Configure your database
+config :healthlocker, Healthlocker.Repo,
+  adapter: Ecto.Adapters.Postgres,
+  username: "postgres",
+  password: "postgres",
+  database: "oxleas_adhd_dev",
```
Comment (SimonLab) [inline apps/healthlocker/config/prod.exs]: Shouldn't we use the prod database on the production configuration file?

File: apps/healthlocker/config/prod.exs [hunk 4]
```diff
@@ -53,9 +74,6 @@ config :logger, level: :info
...
 # start per endpoint:
 #
-#     config :oxleas_adhd, OxleasAdhd.Endpoint, server: true
+#     config :healthlocker, Healthlocker.Endpoint, server: true
 #
-
-# Finally import the config/prod.secret.exs
-# which should be versioned separately.
-import_config "prod.secret.exs"
+# import_config "prod.secret.exs"
```
Comment (SimonLab) [inline apps/healthlocker/config/prod.exs]: Should we uncomment this line to use the production database?

File: apps/healthlocker/config/prod.secrets.exs [hunk 5]
```diff
@@ -0,0 +1,21 @@
+use Mix.Config
```
Comment (SimonLab) [inline apps/healthlocker/config/prod.secrets.exs]: I'm not sure if this file is still use as it is not anymore imported int the config prod

File: apps/healthlocker/app.json [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      "required": true
+    },
+    "SECRET_KEY_BASE": {
+      "generator": "secret"
+    }
+  },
+  "formation": {
+  },
+  "addons": [
+    "heroku-postgresql:hobby-dev"
```
Comment (SimonLab) [inline apps/healthlocker/app.json]: Do we still use the heroku postgres database?

File: apps/healthlocker/demos.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
+# Demos
```
Comment (SimonLab) [inline apps/healthlocker/demos.md]: Do we want to keep this file?

File: apps/healthlocker/mix.exs [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+     {:segment, github: "tonydaly/analytics-elixir"},
+     {:bamboo, "~> 0.7"},
+     {:bamboo_smtp, "~> 1.2.1"},
+     {:mock, "~> 0.2.0", only: :test},
+     {:faker, "~> 0.7"},
+     {:appsignal, "~> 1.0"},
+     {:wallaby, "~> 0.16.1"},
+     {:ecto_factory, "~> 0.0.6"},
+     {:edeliver, "~> 1.4.1"},
+     {:oxleas_adhd, in_umbrella: true},
```
Comment (SimonLab) [inline apps/healthlocker/mix.exs]: :+1:

File: apps/healthlocker/test/controllers/github_version_controller_test.exs [hunk 9]
```diff
@@ -0,0 +1,11 @@
+defmodule Healthlocker.GithubVersionControllerTest do
+  use Healthlocker.ConnCase
+
+  test "GET /_version", %{conn: conn} do
+    {rev, _} = System.cmd("git", ["rev-parse", "HEAD"])
+    version = String.replace(rev, "\n", "")
+    IO.puts version
```
Comment (SimonLab) [inline apps/healthlocker/test/controllers/github_version_controller_test.exs]: can this IO.puts be removed?

File: apps/healthlocker/web/controllers/button_controller.ex [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+  def index(conn, _params) do
+    # for now this only uses one button, so the same image is loaded on error
+    # in the future, the case would try read a file that's put in as a param,
+    # and if it didn't exist it would load the default.
+    button =
+      case File.read("web/static/assets/images/epjs_button.png") do # won't be a dog
+        {:ok, binary} ->
+          Base.encode64(binary)
+        {:error, _} ->
+          binary = File.read!("web/static/assets/images/epjs_button.png")
```
Comment (SimonLab) [inline apps/healthlocker/web/controllers/button_controller.ex]: I think this path is broken. maybe try with ```apps/healthlocker/web/static/assets/images/epjs_button.png```
I'm not sure what is the role of this controller. The index.html only display the image, do we still need this?

**Comment by katbow** [comment]
If it's non blocking, then let's merge so that it's set for Rob and/or I too do the deployment. Thanks so much @SimonLab !

**Pull request merged**